apps/sms/services/README.md

```diff
@@ -0,0 +1,91 @@
+# The layers in the Messages application
+
+## Terms explanation
+
+There are 3 layers whose interfaces managed by the Bridge (aka [threads.js](https://github.com/gaia-components/threads/)):
+
+* Services define high-level semantics to access the model. They are the only
+  interfaces that can be directly called by the views. Ideally one user action
+  has one matching service method.
+
+  Ideally a service doesn't call another service and calls only managers.
+  (but we can relax this rule in our case).
+
+  All together they form the Service Layer.
+
+* Managers define lower-level semantics. They'll be called by the Service layer.
+  Ideally their responsabilities are well-defined and they don't call each other.
+
+* Wrappers define interfaces to access mozAPI that are not available in workers.
+  Ideally they expose very similar APIs than their mozAPI counterparts because
+  their goal is to eventually disappear.
+
+## The Service Layer
+### ConversationService
+
+This service manages everything related to directly interacting with
+conversations and messages.
+
+Here are the operations needed:
+
+* Get All Conversations
+  - used in InboxView
+  - contains everything needed to display the Inbox (so including the "body", the "type", the "draft" and the "contact information".
+* Delete Conversations
+  -  used in InboxView
+* Delete Messages
+  -  used in ConversationView (and can in turn delete Conversations)
+* Mark Conversations As Read/Unread
+  -  Used in InboxView and ConversationView
+* Load 1 Conversation + its messages + contact(s) information(s)
+  -  Used in ConversationView
+* send an event when a message change and the local IDB (if existing) has been updated.
+* send an event when receiving a message and the local IDB (if existing) has been updated.
+* Load 1 Message
+  -  Used in ReportView and GroupView
+  - (If these views are really part of the ConversationPanel, maybe we don't need this operation)
+* find a conversation from number
+  - (if the redirection (new message -> conversation) is done from the service worker, this will be used from the service worker)
+
+### MessagingService
+
+This service will be used when a view does actions related to messaging functionality,
+especially sending messages.
+
+Here are the operations for this service:
+
+* send a message
+  - used from ConversationView and NewMessageView
+  - this service may use other services (eg: ConversationService to change model --
+    or underlying Manager ? or same JS file in both services that could do this ?)
+* the various events coming from mozMobileMessage
+* onSmsReceived (not an event, an actual method that will be called when we get
```

Comment:
Cant it be some sort of ```SmsReceivedService``` that sits inside window (view window context and smth other, not sure) that can handle ```sms-received``` system message?

When it receives system message, it broadcasts event ```sms-received``` via ```bridge.service.broadcast``` and ```MessagingService``` just subscribes to this event _internally_ via ```bridge.client.on``` (maybe with ```OClient``` :)). And then ```MessagingService``` emits to the view the same event as it would when ```mozMobileMessage.onsmsreceived``` triggered.